test/Microsoft.AspNet.Mvc.Core.Test/SessionStateTempDataProviderTest.cs

```diff
@@ -87,12 +226,63 @@ private HttpContext GetHttpContext(ISessionCollection session, bool sessionEnabl
             {
                 httpContext.Setup(h => h.Session).Throws<InvalidOperationException>();
             }
+            else
+            {
+                httpContext.Setup(h => h.Session[It.IsAny<string>()]);
+            }
             if (sessionEnabled)
             {
                 httpContext.Setup(h => h.GetFeature<ISessionFeature>()).Returns(Mock.Of<ISessionFeature>());
-                httpContext.Setup(h => h.Session[It.IsAny<string>()]);
             }
             return httpContext.Object;
         }
+
+        private class TestSessionCollection : ISessionCollection
```

Comment:
@Tratcher, is there an easy way to re-use `SessionCollection` for test purposes?